src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Block.cs

```diff
@@ -19,13 +20,24 @@ public override BoundNode VisitBlock(BoundBlock node)
             }
 
             var builder = ArrayBuilder<BoundStatement>.GetInstance();
-
+            var usingDeclarationIndicies = ArrayBuilder<int>.GetInstance();
             for (int i = 0; i < node.Statements.Length; i++)
             {
-                var stmt = (BoundStatement)Visit(node.Statements[i]);
-                if (stmt != null) builder.Add(stmt);
+                BoundStatement statement = node.Statements[i];
+                if (statement.ContainsUsingDeclarationStatement())
+                {
+                    usingDeclarationIndicies.Add(i);
```

Comment:
This is more complicated than I expected. Could we do this all in one pass by exploiting recursion to visit every statement range starting after the `using`?